### PR TITLE
Frontend Build Script Refactor

### PR DESCRIPTION
**Pull request by adamrdrew**
This patch refactors the frontend build scripts in the following ways:

- Start using the cicd-tools library to make the scripts safer to use and test, including working locally
- Make the code cleaner and easier to read via function decomposition
- Get rid of cruft, commented out old stuff, unused variables, etc
- Try to better manage scope
- Consistency in names and style
- Try to eliminate vectors for bugs and things that are unclear

### CODE REVIEW

File: src/frontend-build.sh [hunk 1]
```diff
@@ -1,240 +1,216 @@
...
 
-  # Build and push the aggregated image
-  # This image is tagged with just the SHA for the current build
-  # as this is the one we want deployed
-  docker  build --label "image-type=aggregate" -t "${IMAGE}:${IMAGE_TAG}" $APP_ROOT -f $APP_ROOT/Dockerfile
-  docker  push "${IMAGE}:${IMAGE_TAG}"
+main() {
+  # Load the CICD helper scripts
+  load_cicd_helper_functions container
+  load_cicd_helper_functions image_builder
```
Comment (Victoremepunto) [inline src/frontend-build.sh:197]: image_builder already loads `container` so you can leave `image_builder` alone

Comment (adamrdrew) [reply src/frontend-build.sh:202]: Removed in recent push

**Comment by adamrdrew** [comment]
I think this is ready for full review. I ran it locally and it mostly worked - the build depends on a specific directory structure so I ran into problems with the final docker builds because my directory layout wasn't up to snuff. but that also shouldn't be super important because by the time we've made it that far in the process we should be all good. that said I'm open to suggestions to make it less dependent on the specific directory structure on the jenkins node.

**Review by Victoremepunto — CHANGES_REQUESTED** [review]

File: src/frontend-build.sh [hunk 2]
```diff
@@ -1,240 +1,216 @@
...
-    CONTAINER_NAME="${APP_NAME}-pr-check-${ghprbPullId}-${timestamp}"
-  fi
-
-  if [ ! -z "$gitlabMergeRequestIid" ]; then
-    export IMAGE_TAG="pr-${gitlabMergeRequestIid}-${IMAGE_TAG}"
-    CONTAINER_NAME="${APP_NAME}-pr-check-${gitlabMergeRequestIid}-${timestamp}"
+build_and_push_pr_image() {
+  # Guard clause to ensure this function is for PR builds
+  if [ "$IS_PR" != true ]; then
+    return
```
[the remark this answers is not in the document]
Comment (adamrdrew) [reply src/frontend-build.sh]: I think I addressed this in the most recent push. Let me know if I got it right.

File: src/frontend-build.sh [hunk 3]
```diff
@@ -1,240 +1,216 @@
...
   fi
 
-  IS_PR=true
-fi
-
-set -ex
-
+  local default_tag=$(cicd::image_builder::get_image_tag)
+  export BUILD_CONTEXT="APP_ROOT"
+  export ADDITIONAL_TAGS=("${default_tag}")
```
[the remark this answers is not in the document]
Comment (adamrdrew) [reply src/frontend-build.sh]: I think I addressed this in the most recent push. Let me know if I got it right.

File: src/frontend-build.sh [hunk 4]
```diff
@@ -1,240 +1,216 @@
...
+    cd "$WORKSPACE/build/container_workspace/"
+    export APP_ROOT="$WORKSPACE/build/container_workspace/"
 
-# source <(curl -sSL $COMMON_BUILDER/src/quay_push.sh | bash -s)
+    # If this is a PR build, append expiration label to Dockerfile
+    if [ "$IS_PR" = true ]; then
+        echo -e "\nLABEL quay.expires-after=3d" >> "$APP_ROOT/Dockerfile"
+    else
+        echo "Publishing to Quay without expiration"
+    fi
```
[the remark this answers is not in the document]
Comment (adamrdrew) [reply src/frontend-build.sh]: I think I addressed this in the most recent push. Let me know if I got it right.

File: src/frontend-build.sh [hunk 5]
```diff
@@ -1,240 +1,216 @@
...
 
-# IMAGE, IMAGE_TAG, and Tokens are exported from upstream pr_check.sh
-export LC_ALL=en_US.utf-8
-export LANG=en_US.utf-8
-export APP_ROOT=${APP_ROOT:-pwd}
-export WORKSPACE=${WORKSPACE:-$APP_ROOT}  # if running in jenkins, use the build's workspace
-# export IMAGE_TAG=$(git rev-parse --short=7 HEAD)
-export GIT_COMMIT=$(git rev-parse HEAD)
+    # Get the latest git commit hash
+    export GIT_COMMIT=$(git rev-parse HEAD)
```
[the remark this answers is not in the document]
Comment (adamrdrew) [reply src/frontend-build.sh]: I think I addressed this in the most recent push. Let me know if I got it right.

File: src/frontend-build.sh [hunk 6]
```diff
@@ -1,240 +1,216 @@
...
+  
+  if [ -n "$ghprbPullId" ]; then
+    pr_id=$ghprbPullId
+  elif [ -n "$gitlabMergeRequestIid" ]; then
+    pr_id=$gitlabMergeRequestIid
+  fi
 
-  # Get the the last 6 builds
-  getHistory
+  export IMAGE_TAG="pr-${pr_id}-${IMAGE_TAG}"
```
[the remark this answers is not in the document]
Comment (adamrdrew) [reply src/frontend-build.sh]: I think I addressed this in the most recent push. Let me know if I got it right.

File: src/frontend-build.sh [hunk 7]
```diff
@@ -1,240 +1,216 @@
...
+  export CONTAINER_NAME="${APP_NAME}-pr-check-${pr_id}-${timestamp}"
+  export IS_PR=true
```
[the remark this answers is not in the document]
Comment (adamrdrew) [reply src/frontend-build.sh]: I think I addressed this in the most recent push. Let me know if I got it right.

**Review by wcmitchell — APPROVED** [review]
Ran through this today, just for an extra set of eyes. Lots of change, but nothing standing out to me as an issue. 👍

**Review by Victoremepunto — COMMENTED** [review]
LGTM - I'm sorry I didn't submit this (I thought I did) I think there's further improvement worth of a follow up - I can work on it too if you want. 

Great job btw!

**Pull request merged**